docs/fast_data/configuration/projection_storer.md

```diff
@@ -0,0 +1,1048 @@
+their own file. When the files containing the _user-defined functions_ are loaded, the service will search within them
+for a function named as the key name in the configuration. The function with such name **must** exist otherwise the service will
+encounter a processing error.   
+
+The `castFunctions` property is then configurable as follows: it should be a mapping between names of custom cast functions
+and the path on the service where to find the file containing its implementation.  
+Here it is shown a possible example of configuring two custom cast functions:
+
+```json
+"castFunctions": {
+  "mapToAddressType": "/app/extensions/mapToAddressType.kts",
+  "castToTitleCase": "/app/extensions/castToTitleCase.js"
+}
+```
+
+:::note
+It is important to remember to mount each custom cast function or the folder containing them within the plugin instance.
+Then in the `castFunctions` property they should be properly named and set the correct location to each file containing
+the function's implementation. 
+:::
+
+Considering the implementation of these cast functions, they expect as input two parameters, that are the _value_ the field
+to be transformed and the _field name_ represented as `string`. The output of the cast functions should be a single value
+in the type expected by the data model for that specific field on which the cast function is applied.
+
+Below is provided an example of cast functions implementation, one for each supported programming language.
+
+<details><summary>Custom Cast Function (Kotlin - mapToAddressType.kts)</summary>
+<p>
+
+```kotlin
+package castFunctions
+
+val addressMapping = mapOf(
+  1 to "SHIPPING",
+  2 to "BILLING",
+  3 to "LIVING",
+)
+
+// NOTE: the name of the function must correspond to
+//       the key associated to the file containing it
+fun mapToAddressType(value: Any, fieldName: String): String? {
+  return when (value) {
+    is String -> addressMapping[value.toInt()]
+    is Int, is Long -> addressMapping[value]
+    else -> {
+      // NOTE: a basic logger can be accesses via internal binding
+      logger.debug("not an address type code: $value - fieldName: $fieldName")
+      
+      null
+    }
+  }
+}
+```
+
+</p>
+</details>
+
+<details><summary>Custom Cast Function (Javascript - castToTitleCase.js)</summary>
+<p>
+
+```javascript
+'use strict'
+
+// NOTE: the name of the function must correspond to
+//       the key associated to the file containing it
+function castToTitleCase(value, fieldName) {
+  const str = value.toString()
+  
+  str[0].toUpperCase() + str.slice(1).toLowerCase()
+}
+```
+
+</p>
+</details>
+
+:::danger
+For those migrating custom cast functions from the ones employed in the Real-Time Updater, please bear in mind that
+the `logger` parameter is not provided anymore in the function signature, and therefore it should be removed.  
+For debugging purposes, a logger is still provided via internal bindings, but it is recommended to not log
```

Comment:
Maybe adding it into the example with a note, I don't really know what do you mean by internal bindings, should I import something, is the object globally available, inside the `this`?

Reply:
Also it could be nice to link the doc to the provided logger, like pino in case of JS

Reply:
Added logger also in JS example